### ISSUE TITLE
`foo` is valid JSON but it shouldn't

### ISSUE THREAD

**Issue by amtoine** [nushell/nushell#10606]
### Describe the bug

 as title.

### How to reproduce

1. run
```nushell
"'foo'" | from json
```
2. run
```nushell
"foo" | from json
```

### Expected behavior

because `foo` is not `"foo"` nor `'foo'` and thus invalid JSON, `from json` should give an error.

### Configuration

| key                | value                                    |
| ------------------ | ---------------------------------------- |
| version            | 0.85.1                                   |
| branch             | main                                     |
| commit_hash        | 8c507dc984693c45b0440adf63a94e0410b4bc42 |
| build_os           | linux-x86_64                             |
| build_target       | x86_64-unknown-linux-gnu                 |
| rust_version       | rustc 1.70.0 (90c541806 2023-05-31)      |
| rust_channel       | 1.70.0-x86_64-unknown-linux-gnu          |
| cargo_version      | cargo 1.70.0 (ec8a8a0ca 2023-04-25)      |
| build_time         | 2023-10-04 18:35:49 +02:00               |
| build_rust_channel | release                                  |
| allocator          | mimalloc                                 |
| features           | default, sqlite, trash, which, zip       |
| installed_plugins  | gstat, nu_plugin_explore                 |

**Comment by makeittotop** [comment]
@amtoine HI! Would it be OK if I give this one a go?

**Comment by amtoine** [comment]
@makeittotop 
i was not working on this, please go ahead and make Nushell better :wink: :pray:

**Comment by makeittotop** [comment]
Thanks @amtoine . If it's possible, would you be able to point me to a particular direction to help me get started? The repo. as I had correctly imagined is a little bit evolved for a Rust n00b like me.

**Comment by sholderbach** [comment]
Mhh, that goes quickly into the details of [`nu-json`](https://github.com/nushell/nushell/tree/main/crates/nu-json) our crate to perform a more relaxed JSON parsing.
https://github.com/nushell/nushell/blob/c925537c48ffa2bd927dd33f30af462483fea657/crates/nu-command/src/formats/from/json.rs#L151-L181

It takes liberties following [hjson](https://hjson.github.io/) to allow comments, trailing commas etc.
`nu-json` is apparently a fork of [`serde-hjson`](https://crates.io/crates/serde-hjson).

If I read their spec it seems [hjson allows unquoted strings in the root](https://hjson.github.io/syntax.html).

So the question becomes if we want to allow that behavior for convenience or if we should be more strictly following the original [JSON spec](https://www.ecma-international.org/publications-and-standards/standards/ecma-404/) by default.

**Comment by amtoine** [comment]
@makeittotop 
yeah i've never looked at the `from json` command nor the `nu-json` crate, so you might need to dig around a bit :relieved: 

of course there to help if need be :wink:

**Comment by Sheape** [comment]
Hi, I'd like to believe this is resolved now? By using `from json --strict` which uses `serde_json` instead of `nu_json`, this follows the JSON specification exactly as it is. `from json` still uses `nu_json` in case you'd want this behavior for convenience. So you get the best of both worlds 😊

**Comment by cptpiepmatz** [comment]
As @Sheape mentioned, this is fixed now via `--strict´.
